baseapp/baseapp.go

```diff
@@ -954,6 +960,16 @@ func (app *BaseApp) runTx(mode execMode, txBytes []byte) (gInfo sdk.GasInfo, res
 		result, err = app.runMsgs(runMsgCtx, msgs, msgsV2, mode)
 	}
 
+	if mode == execModeSimulate {
+		nestedMsgsContext, _ := app.cacheTxContext(ctx, txBytes)
+		for _, msg := range msgs {
+			nestedErr := app.simulateNestedMessages(nestedMsgsContext, msg)
```

Comment:
nested messages may be executed at a later date, this would cause higher gas to be returned to the user in the case of the simulation.

We should look over the sdk and see what should consume gas within the tx simulation vs something that should not. For example a gov proposal simulation should not return higher gas than what's needed for the proposal message. Now it seems it would show message and nested message consumption, which could lead to the user paying more in fees.

Reply:
The gas is not currently being added but true. Sometimes we'd like to add the gas and others just check the messages are valid. 🤔🤔

About gov, not sure how useful this may be. For example, the validation of a simple proposal with a send may fail because the gov module address doesn't have the funds to fulfil that send.